src/server/socket.test.ts

```diff
@@ -779,5 +780,251 @@ describe("Server-side socket testing", () => {
         return Promise.all([admin_remove_callbacks, admin_remove_events]);
       });
     });
+
+    describe("Grasp reactions", () => {
+      type Count = {
+        level: "good" | "unsure" | "lost";
+        count: number;
+      };
+
+      beforeEach(async () => {
+        // Mocked timestamp
+        jest.spyOn(Date, "now").mockImplementation(() => 1673942400000);
+
+        const [{ id: roomId }] = await knex("Room")
+          .select("id")
+          .where("visibleId", "a418c099-4114-4c55-8a5b-4a142c2b26d1");
+
+        const [{ id: userId }] = await knex
+          .table("AnonGraspUser")
+          .insert({})
+          .returning("*");
+
+        // Must be in order of most recent to least recent
+        // Assumption is reactions will be added in chronological order
+        const graspReactions = [
+          {
+            level: "lost",
+            sent_at: new Date(Date.now() - 5 * 60 * 1000),
+          },
+          {
+            level: "good",
+            sent_at: new Date(Date.now() - 3 * 60 * 1000),
+          },
+          {
+            level: "unsure",
+            sent_at: new Date(Date.now()),
+          },
+        ];
+
+        for (const graspReaction of graspReactions) {
+          await knex("GraspReaction").insert(
+            {
+              level: graspReaction.level,
+              sent_at: graspReaction.sent_at,
+              room_id: roomId,
+              is_active: true,
+              anon_grasp_user_id: userId,
+            },
+            ["*"],
+          );
+        }
+      });
+
+      afterEach(() => {
+        jest.spyOn(Date, "now").mockRestore();
+      });
+
+      test("Data based on active grasp reactions are sent on connection to admin", async () => {
+        const { address, port } = server.address() as AddressInfo;
+        socket_admin = Client(
+          `http://[${address}]:${port}/rooms/a418c099-4114-4c55-8a5b-4a142c2b26d1/admin`,
+          {
+            autoConnect: false,
+          },
+        );
+
+        const events = allEvents(socket_admin, [
+          "GraspReactionGet",
+          "connect",
+          "RoomJoined",
+        ]).then((value: unknown[]) => {
+          const graspReactionGetResults = value[0];
+          const counts = graspReactionGetResults as Count[];
+
+          counts.forEach((entry) => {
+            expect(typeof entry.level).toBe("string");
+            expect(typeof entry.count).toBe("string");
```

Comment:
I think it had something to do with the count calculations taking place in the database. Since I've changed it so that the calculations happen in JS, it no longer expects strings but numbers instead.